### PR TITLE
CB-315: Show top contributors

### PR DESCRIPTION
**Pull request by FannyVieira**
Added database query to get top 10 contributors from the last week

### CODE REVIEW

**Comment by code-master5** [comment]
@FannyVieira 
Thanks for this PR.

However, I would like to tell you that the ticket for which you've opened this PR has already been assigned to someone else. It's a good habit to ask the assignee first. There is quite a possibility that they've already started working on the issue.
Again, it's a good habit to ask you mentor @paramsingh (IRC Nick: iliekcomputers) first about what kind of results they expect regarding the ticket on which you plan to work.
These are some generic points to consider before opening a PR on any MetaBrainz project.

File: critiquebrainz/db/review.py [hunk 1]
```diff
@@ -684,3 +684,28 @@ def reviewed_entities(*, entity_ids, entity_type):
...
+            "user".musicbrainz_id,
+            "user".is_blocked
+        FROM review JOIN "user" 
+        ON review.user_id = "user".id
+        WHERE review.published_on > current_date - interval '7 days'
+        AND review.is_draft <> TRUE
+        GROUP BY review.user_id, "user".id
+        ORDER BY qtd DESC
+        LIMIT :limit
+    """)
```
Comment (code-master5) [inline critiquebrainz/db/review.py]: Please follow the generic indentation guidelines for SQL queries as per project preference. This applies to files, function, classes, dictionaries, and every other thing. 
For SQL queries, in particular, you can have a look at other SQL queries.
If you feel that your way is better than the current way, please discuss that with your mentor before moving on.

File: critiquebrainz/db/review.py [hunk 2]
```diff
@@ -684,3 +684,28 @@ def reviewed_entities(*, entity_ids, entity_type):
         })
         reviewed_ids = [str(row[0]) for row in results.fetchall()]
     return reviewed_ids
+
+def get_top_contributors(limit=10):
+    query = sqlalchemy.text("""
```
Comment (code-master5) [inline critiquebrainz/db/review.py]: We prefer docstrings for any function we add to the codebase. Have a look at other functions for your reference.

File: critiquebrainz/db/review.py [hunk 3]
```diff
@@ -684,3 +684,28 @@ def reviewed_entities(*, entity_ids, entity_type):
...
+        AND review.is_draft <> TRUE
+        GROUP BY review.user_id, "user".id
+        ORDER BY qtd DESC
+        LIMIT :limit
+    """)
+    with db.engine.connect() as connection:
+        results = connection.execute(query, {"limit": limit})
+        rows = results.fetchall()
+        return rows
+
```
Comment (code-master5) [inline critiquebrainz/db/review.py]: Again here I guess you should not be directly returning the results.
It's a good programming practice to consider cases like:
What happens if db is not able to fetch any result?
What is the `type` of result? Do we need to convert results to some suitable `type` before returning them?
And so on.

**Comment by code-master5** [comment]
@FannyVieira 
One more thing.
Please open a separate PR for any feature or bug fix. This PR is committed over your previous PR which is still waiting to be merged.

@paramsingh Your comments please!

File: critiquebrainz/db/review.py (hunk 1 above)
Comment (paramsingh) [reply critiquebrainz/db/review.py]: We have some style guildelines documented here: https://github.com/metabrainz/guidelines/blob/master/Python.md

Specifically
>SQL keywords should be in upper-case and right aligned.

File: critiquebrainz/db/review.py (hunk 2 above)
Comment (paramsingh) [reply critiquebrainz/db/review.py]: The docstrings we use are google style docstrings, documented here: https://github.com/metabrainz/guidelines/blob/master/Python.md#docstrings

**Comment by paramsingh** [comment]
Hi @FannyVieira!

Thanks for the contribution! However as @code-master5 has mentioned, this issue was already being worked on by another person, that's why it was assigned to them on JIRA. It would be great if you could make sure that no one is working on an issue before taking it. This generally isn't a big problem, we're a small team, but it does cause problems when it happens. 

Also, note that this PR just adds an SQL query with no way to see the results on the website or the webservice. We generally try to do small but complete PRs. For example, here a complete PR would have used the function you added to get some data into the frontend module and show it in the flask templates. 

This particular issue was being worked on by @code-master5 and I'm not sure how much progress they've made but it would be great if we could coordinate on IRC to determine who's working on this issue. There isn't a shortage of issues to fix and it'd be great if no work is duplicated. 

Thanks for understanding. =)

**Comment by FannyVieira** [comment]
> @FannyVieira
> Thanks for this PR.
> 
> However, I would like to tell you that the ticket for which you've opened this PR has already been assigned to someone else. It's a good habit to ask the assignee first. There is quite a possibility that they've already started working on the issue.
> Again, it's a good habit to ask you mentor @paramsingh (IRC Nick: iliekcomputers) first about what kind of results they expect regarding the ticket on which you plan to work.
> These are some generic points to consider before opening a PR on any MetaBrainz project.

I'm sorry, I did not see that I already had a person assigned to this task, I also apologize for not following the standard step-by-step of the community, I will take that care of the next times

File: critiquebrainz/db/review.py (hunk 2 above)
Comment (FannyVieira) [reply critiquebrainz/db/review.py]: Okay, thank u for the references, i've added. :smile:

File: critiquebrainz/db/review.py (hunk 1 above)
Comment (FannyVieira) [reply critiquebrainz/db/review.py]: Fixed. :smile_cat:

File: critiquebrainz/db/review.py (hunk 3 above)
Comment (FannyVieira) [reply critiquebrainz/db/review.py]: I understand, i had not stopped to think about what you said, but it makes sense

**Comment by FannyVieira** [comment]
> @FannyVieira
> One more thing.
> Please open a separate PR for any feature or bug fix. This PR is committed over your previous PR which is still waiting to be merged.
> 
> @paramsingh Your comments please!

Alright, sorry for this confusion :confounded:

**Comment by FannyVieira** [comment]
> Hi @FannyVieira!
> 
> Thanks for the contribution! However as @code-master5 has mentioned, this issue was already being worked on by another person, that's why it was assigned to them on JIRA. It would be great if you could make sure that no one is working on an issue before taking it. This generally isn't a big problem, we're a small team, but it does cause problems when it happens.
> 
> Also, note that this PR just adds an SQL query with no way to see the results on the website or the webservice. We generally try to do small but complete PRs. For example, here a complete PR would have used the function you added to get some data into the frontend module and show it in the flask templates.
> 
> This particular issue was being worked on by @code-master5 and I'm not sure how much progress they've made but it would be great if we could coordinate on IRC to determine who's working on this issue. There isn't a shortage of issues to fix and it'd be great if no work is duplicated.
> 
> Thanks for understanding. =)

Oh, yes, I apologize for the confusion I did not see that had already been attributed to one person and in the next few times I will follow the guidelines that were passed to me.

**Comment by FannyVieira** [comment]
Hi, do i can work on this bug, i was waiting for some feedback

**Pull request closed without merging**